Client/Application/AppDelegate.swift

```diff
@@ -364,6 +358,22 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         case .NewPrivateTab:
             browserViewController.applyPrivateModeTheme(force: true)
             browserViewController.openBlankNewTabAndFocus(isPrivate: true)
+        case .OpenLastBookmark:
+            // open bookmark in a non-private browsing tab
+            browserViewController.applyNormalModeTheme(force: true)
+            // find out if bookmarked URL is currently open
+            // if so, open to that tab,
+            // otherwise, create a new tab with the bookmarked URL
+            if let urlString = userData?["bookmarkURL"] as? String,
+                let urlToOpen = NSURL(string: urlString) {
+                let index = browserViewController.tabManager.tabs.indexOf { $0.url == urlToOpen }
+                if let index = index {
+                   let bookmarkTab = browserViewController.tabManager.tabs[index]
+                    browserViewController.tabManager.selectTab(bookmarkTab)
```

Comment:
Nit: Since we're reaching a few levels deep into browserViewController to get the tab index and select, we should instead refactor this functionality into the tabManager:
```swift
func selectTabWithURL(url: NSURL) -> Browser? {
    guard let index = tabs.indexOf { $0.url == url }
          let tab = tabs[index] else { return nil }
    return selectTab(tab)
}
```